core/amber/src/main/python/core/architecture/managers/channel_marker_manager.py

```diff
@@ -0,0 +1,65 @@
+from collections import defaultdict
+from typing import Set, Dict
+
+from proto.edu.uci.ics.amber.core import ActorVirtualIdentity, ChannelIdentity
+from proto.edu.uci.ics.amber.engine.architecture.rpc import (
+    ChannelMarkerPayload,
+    ChannelMarkerType,
+)
+
+
+class ChannelMarkerManager:
+    def __init__(self, actor_id: ActorVirtualIdentity, input_gateway):
+        self.actor_id = actor_id
+        self.input_gateway = input_gateway
+        self.marker_received: Dict[str, Set[ChannelIdentity]] = defaultdict(set)
+
+    def is_marker_aligned(
+        self, from_channel: ChannelIdentity, marker: ChannelMarkerPayload
```

Comment:
your PR description did not talk anything about the Channel marker manager, which I think is a core change of this PR?

Reply:
revised the PR description.